scripts/windows_dev_setup.ps1

```diff
@@ -336,17 +360,133 @@ function existing_package {
 	Write-Host "This package is already installed."
 }
 
-welcome_message
-verify_architecture
-check_os
-install_msvc_build_tools
-install_llvm
-install_openssl
-install_nodejs
-install_pnpm
-install_postgresql
-install_python
-install_protoc
-install_rustup
-install_cargo_plugins
+function install_git {
+  if (Get-Command git -ErrorAction SilentlyContinue) {
+    Write-Host "Installing Git..."
+    winget install Git.Git --silent
+  } 
+  else {
+    winget upgrade --id Git.Git
+  }
+}
+
+function install_cvc5 {
+  if (![System.IO.Path]::IsPathRooted($env:CVC5_EXE)) {
+    Write-Host "Installing CVC5..."
+    # Download and extract the 64-bit version of Protoc
+    $cvc5_exe = "$env:USERPROFILE\cvc5-$global:cvc5_version\cvc5-Win64.exe"
+    Write-Host $path
+    Invoke-WebRequest -Uri "https://github.com/cvc5/cvc5/releases/download/cvc5-$global:cvc5_version/cvc5-Win64.exe" -OutFile (New-Item -Path "$cvc5_exe" -Force) -ErrorAction SilentlyContinue
+    [Environment]::SetEnvironmentVariable("CVC5_EXE", "$cvc5_exe", "User") 
+    Write-Host "User environment variables set for CVC5"
+  }
+  elseif ($global:architecture -eq "86") {
+    Write-Host "Only 64-bit systems can install CVC5"
+  }
+  else {
+    Write-Host "CVC5 is already installed."
+  }
+}
+
+function install_dotnet {
+  if (![System.IO.Path]::IsPathRooted($env:DOTNET_ROOT)) {
+    Write-Host "Installing Microsoft DotNet..."
+    winget install "Microsoft.DotNet.SDK.6" --accept-source-agreements --silent
+    [Environment]::SetEnvironmentVariable("DOTNET_ROOT", "$env:PROGRAMFILES\dotnet", "User")
+    [Environment]::SetEnvironmentVariable("PATH", "$env:PATH;$env:PROGRAMFILES\dotnet\sdk\$global:dotnet_version\DotnetTools;;$env:PATH;$env:USERPROFILE\.dotnet\tools", "User")
+    Write-Host "User environment variables set for DotNet"
+  } 
+  else {
+    Write-Host "Microsoft DotNet is already installed."
+  }
+}
+
+function install_z3 {
+  if (![System.IO.Path]::IsPathRooted($env:Z3_EXE)) {
+    Write-Host "Installing Z3..."
+    $uri = "z3-$global:z3_version"
+    $z3_zip = "z3-$global:z3_version-x$global:architecture-win.zip"
+    $z3_filepath = "$env:USERPROFILE\$z3_zip"
+    # Download and extract the 64-bit version of Protoc
+    Invoke-WebRequest -Uri "https://github.com/Z3Prover/z3/releases/download/$uri/$z3_zip" -OutFile (New-Item -Path "$z3_filepath" -Force) -ErrorAction SilentlyContinue
+    if ($global:architecture -eq "64") {
+      while ((Get-Item "$z3_filepath").Length -lt 52MB) {
+        Start-Sleep -Seconds 1
+      }
+    }

```

Comment:
This is a download and extract.  Maybe we want to print a `.` or something so there's progress as the user is waiting for the download.  A quick progress bar kind of thing.

I also wonder if there's a way to get these file sizes before downloading, since the size could change over time / this could get out of sync

Reply:
@gregnazario, PowerShell already has a built-in progress bar which automatically displays for the `Invoke-WebRequest` and `Expand-Archive` commands. It looks like this:
![image](https://user-images.githubusercontent.com/125399153/231278428-ebc95eda-a28a-4814-93d9-b962e409238c.png)

I meant to remove the while loops earlier, as they are unnecessary. `Invoke-WebRequest` and `Expand-Archive` work synchronously by default. They wait for the download/extraction to complete, regardless of size, before moving on to the next step.